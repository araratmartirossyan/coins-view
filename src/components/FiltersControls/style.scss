.filters-controls {
  display: flex;
  width: 100%;

  & .filters-controls__favorite {
    width: auto;
    margin-right: 30px;
  }

  & .filters-controls__selects-group {
    margin-left: auto;
  }

  & .filters-controls__select:not(:last-child) {
    margin-right: 30px;
  }

  & .filters-controls__select {
    min-width: 100px;
  }
}
