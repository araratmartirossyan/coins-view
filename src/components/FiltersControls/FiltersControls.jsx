import React from 'react';

import Box from '@material-ui/core/Box';
import IconButton from '@material-ui/core/IconButton';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';

import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '@material-ui/core/Button';

import BookmarkIcon from '@material-ui/icons/Bookmark';
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';

import './style.scss';

const getClassForButton = (state, value) => {
  return state === value ? "primary" : "default"
}

export default function FiltersControls(props) {
  const {filter, changeFilterValue, toggleShowFavorite, showFavorite} = props

  const handleChangeButton = (event) => {
    const value = event.target.innerText
    changeFilterValue(value)
  };

  const handleChangeSelect = (event) => {
    const value = event.target.value
    changeFilterValue(value)
  };

  return (
    <Box className="filters-controls">
      <IconButton onClick={toggleShowFavorite} className="filters-controls__favorite"  color="secondary" aria-label="favorites">
        {showFavorite ? <BookmarkIcon color="secondary" /> :  <BookmarkBorderIcon color="secondary" />}
      </IconButton>
      <ButtonGroup className="filters-controls__buttons-group" disableElevation size="small" variant="contained">
        <Button color={getClassForButton(filter, 'BNB')} onClick={handleChangeButton}>BNB</Button>
        <Button color={getClassForButton(filter, 'BTC')} onClick={handleChangeButton}>BTC</Button>
      </ButtonGroup>
      <Box className="filters-controls__selects-group">
        <FormControl className="filters-controls__select">
          <InputLabel id="alts-select-label">ALTS</InputLabel>
          <Select
            labelId="alts-select-label"
            id="alts-select"
            value={filter}
            onChange={handleChangeSelect}
          >
            <MenuItem value={'ETH'}>ETH</MenuItem>
            <MenuItem value={'TRX'}>TRX</MenuItem>
            <MenuItem value={'XPR'}>XPR</MenuItem>
          </Select>
        </FormControl>
        <FormControl className="filters-controls__select">
          <InputLabel id="usds-select-label">USDS</InputLabel>
          <Select
            labelId="usds-select-label"
            id="usds-select"
            value={filter}
            onChange={handleChangeSelect}
          >
            <MenuItem value={'USDT'}>USDT</MenuItem>
            <MenuItem value={'EUR'}>EUR</MenuItem>
          </Select>
        </FormControl>
      </Box>
    </Box>
  );
}
