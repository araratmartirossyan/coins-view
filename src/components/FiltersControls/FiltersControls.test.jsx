import React from 'react'
import renderer from 'react-test-renderer'

import FiltersControls from './FiltersControls'

it('renders correctly FiltersControls', () => {
  const input = renderer.create(<FiltersControls />).toJSON()
  expect(input).toMatchSnapshot()
})
