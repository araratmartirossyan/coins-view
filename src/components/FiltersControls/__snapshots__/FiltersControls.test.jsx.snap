// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly FiltersControls 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 filters-controls"
>
  <button
    aria-label="favorites"
    className="MuiButtonBase-root MuiIconButton-root filters-controls__favorite MuiIconButton-colorSecondary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
        />
      </svg>
    </span>
  </button>
  <div
    className="MuiButtonGroup-root filters-controls__buttons-group MuiButtonGroup-disableElevation MuiButtonGroup-contained"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        BNB
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        BTC
      </span>
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-77 filters-controls__selects-group"
  >
    <div
      className="MuiFormControl-root filters-controls__select"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        id="alts-select-label"
      >
        ALTS
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="alts-select-label alts-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="alts-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiFormControl-root filters-controls__select"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        id="usds-select-label"
      >
        USDS
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="usds-select-label usds-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="usds-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
